Cloudera Eucalyptus theme update

pomegranited opened the pull request:
Updates the [Appsembler cloudera-dogwood theme](https://github.com/appsembler/edx-theme/tree/appsembler/cloudera-dogwood) for use with Eucalyptus.

**Sandbox URL**:

* LMS: https://eucalyptus-theme-updates.stage.opencraft.hosting
* CMS: https://studio-eucalyptus-theme-updates.stage.opencraft.hosting

**Testing instructions**:

1. Create a student account on the [ondemand.cloudera.com](https://ondemand.cloudera.com/) site, and register for one of the Free, Self-Paced courses on offer.
1. Compare the theme deployed to the [sandbox](https://eucalyptus-theme-updates.stage.opencraft.hosting) with the theme on [ondemand.cloudera.com](https://ondemand.cloudera.com/) to ensure there are major irregularities.

**Author notes and concerns**:

1. Please be aware that [ondemand.cloudera.com](https://ondemand.cloudera.com/) is the public customer site, so be careful if posting to discussion boards.
1. We haven't moved all the custom templates into standard edX locations; we've only updated the existing files to work with Eucalyptus.  So some Eucalyptus features (e.g. Bookmarks) may be missing entirely.
1. I've enabled [edX Search](http://edx.readthedocs.io/projects/edx-installing-configuring-and-running/en/latest/configuration/edx_search.html) on the sandbox instance, but haven't seen the search indexes update properly yet.  So course searches from the Dashboard return 0 results.

**Reviewers**
- [x] @bradenmacdonald

mtyaka committed:
Updates Cloudera's theme for Eucalyptus

Major changes:

* Converts to comprehensive theme.
* Replaces microsite templatetags with site configuration.
* Fixes hardcoded image paths and static URLs.
* Removes nav_skip blocks.
  These are no longer supported in eucalyptus where the nav skip button
  now always points to #main.

  This means that the nav skip button currently doesn't work correctly
  with this theme; we will have to go through all templates to add a
  <main id="main"> element.
* Removes main_djano.html override.
  This is no longer necessary in eucalyptus.
* Adds Makefile with task to compile sass during theme development.
* Widens the clickable area for the animated User Profile menu button.
* Replaces html version of README with README.md, updated to reflect latest state.

bradenmacdonald commented:
Bugs I noticed:

1. The next/prev links at the bottom of a content unit are different colors than before:
![screen shot 2016-12-29 at 7 53 27 pm](https://cloud.githubusercontent.com/assets/945577/21559403/a3dbde04-ce00-11e6-9977-e5761096b754.png)

2. The navigation bar appearance is not correct:

Expected:
![screen shot 2016-12-29 at 7 54 11 pm](https://cloud.githubusercontent.com/assets/945577/21559406/b7bb1b10-ce00-11e6-85a5-67ab990c2776.png)

Actual:
![screen shot 2016-12-29 at 7 54 16 pm](https://cloud.githubusercontent.com/assets/945577/21559417/d61dada2-ce00-11e6-9660-114888609340.png)

3. Tab underline appearance on the Account Settings page is not correct:

Expected (but with different colors):
![screen shot 2016-12-29 at 8 02 12 pm](https://cloud.githubusercontent.com/assets/945577/21559466/b411fd02-ce01-11e6-8a54-3712b215d00b.png)


Actual:
![screen shot 2016-12-29 at 8 01 33 pm](https://cloud.githubusercontent.com/assets/945577/21559465/aa6184f8-ce01-11e6-994c-d9a12d69eefb.png)

4. Forms on the account settings page do not match the previous theme

Expected:
![screen shot 2016-12-29 at 8 03 01 pm](https://cloud.githubusercontent.com/assets/945577/21559482/d9ed2b6e-ce01-11e6-91ef-b15aa9c0a775.png)

Actual:
![screen shot 2016-12-29 at 8 03 14 pm](https://cloud.githubusercontent.com/assets/945577/21559484/dcd07cfa-ce01-11e6-8d14-f3821f963c63.png)

5. Should we rename `_cloudera-dogwood.scss` ?
6. Please add a README (similar to the deleted one)

Bugs I noticed that we should fix in the future (not introduced by this PR):
* The menu is not really optimized for desktop use. When opening the mobile menu (big blue screen), if you click on the "x" in the top right but your cursor is between the "spokes" of the X, it doesn't close.

bradenmacdonald commented on lms/templates/courseware/legacy_instructor_dashboard.html:
```diff
@@ -10,7 +10,6 @@
 %>
 
 <%block name="pagetitle">${_("Legacy Instructor Dashboard")}</%block>
-<%block name="nav_skip">#instructor-dashboard-content</%block>
```
How does this screenreader shortcut work in Eucalyptus?

pomegranited replied:
There's an [`a.nav-skip`](https://github.com/edx/edx-platform/blob/6d2468d0dcf782a7561ecc61946146d6fae2762f/lms/templates/main_django.html#L31) at the very top of the `main_django.html` template.

bradenmacdonald commented on lms/templates/footer-extra.html:
```diff
@@ -0,0 +1,12 @@
+<%namespace name='static' file='/static_content.html'/>
+
+<!-- Include jQuery -->
+##<script src="${static.url('js/jquery-1.11.2.min.js')}"></script>
```
Optional: While we're here, can we just remove this instead of commenting it out? (And remove the `jquery-1.11.2.min.js` file, and update the comment on line 6 to say "Include jQuery mobile" only)

pomegranited replied:
Done with 86e5fe8.

bradenmacdonald commented on src/common-elements/_discussions.scss:
```diff
@@ -1,22 +1,25 @@
-  &:hover, &:focus {
-    opacity: 0.6;
+    // Have to specify :not:(:disabled) above to win against
+    // _edx-overrides.scss
```
Is there a nicer way to do this, like adding `html body` first? Or is `:not(:disabled)` more efficient? Also, the comment has a typo (extra `:`)

pomegranited replied:
c6ece21 is nicer.. it uses tag names instead of the `not(:disabled)` pseudoclass, here and for the `.sequence-nav-button`s.

Also has the added benefit of fixing the disabled nav button styles too, e.g. at the [very beginning](https://eucalyptus-theme-updates.stage.opencraft.hosting/courses/course-v1:edX+DemoX+Demo_Course/courseware/d8a6192ade314473a78242dfeedfbf5b/edx_introduction/) and the [very end](https://eucalyptus-theme-updates.stage.opencraft.hosting/courses/course-v1:edX+DemoX+Demo_Course/courseware/1414ffd5143b4b508f739b563ab468b7/workflow/) of the course.

pomegranited commented:
@bradenmacdonald  Thank you for your thorough review!  I'm working through your comments, but will have to finish up on Monday.

1. Next/prev links look ok to me?  The disabled buttons are grey, and the enabled ones are darker.  Maybe it's because Eucalyptus lets you sequence-nav between sections, and so the live Dogwood site is showing grey buttons everywhere, for sections with only one unit?
1. *TODO* Nav bar and hover text
1. *TODO* Tab nav on the Account Settings page 
1. e28d1d8 fixes the responsive widths for the Account Settings form.  Also, I chose to use the default Eucalyptus style for the `<select>` elements, since they're prettier than the old grey boxes.
1. 448d6a3 renamed `_cloudera-dogwood.scss` to `_cloudera.scss`. 
1. *TODO*: update README.md
1. *TODO*: make `#site-menu-toggle` wrapper div clickable instead of just the spokes of the animated X, if it's not too hard to fix.

bradenmacdonald commented:
@pomegranited Thanks for those changes!

> Next/prev links look ok to me? The disabled buttons are grey, and the enabled ones are darker. Maybe it's because Eucalyptus lets you sequence-nav between sections, and so the live Dogwood site is showing grey buttons everywhere, for sections with only one unit?

Sorry, my bad. I probably compared a disabled one with an active one :-/

pomegranited commented:
@bradenmacdonald Ok, I think I've addressed everything now.  What do you think?

1. Next/prev links are ok.
1. ab337e2 Courseware nav bar and hover text, and course navigation menu.
1. 499cfb6 addresses tab nav on the Account Settings page 
1. e28d1d8 fixes the responsive widths for the Account Settings form.  Also, I chose to use the default Eucalyptus style for the `<select>` elements, since they're prettier than the old grey boxes.
1. 448d6a3 renamed `_cloudera-dogwood.scss` to `_cloudera.scss`. 
1. ff4f135 Updates README.md
1. 56d8a58: makes `#site-menu-toggle` wrapper div clickable instead of just the spokes of the animated X.

bradenmacdonald commented:
@pomegranited Thanks! Looking good but I noticed a few more things:

> 56d8a58: makes #site-menu-toggle wrapper div clickable instead of just the spokes of the animated X.

8. Now, clicking anywhere, including on a nav menu link, closes that blue popover. "Close by clicking anywhere" is perhaps ok, but seems weird. Whether you meant that or not, clicking the nav links should not close that menu.

9. Account Settings form looks nice, but there's an issue when making any change:
![screen shot 2017-01-02 at 5 36 26 pm](https://cloud.githubusercontent.com/assets/945577/21598472/7e148864-d112-11e6-8861-9924910a48ab.png)

10. Minor: the animations when hovering over/off of the next/prev navbar buttons above the courseware are very weird (in Chrome at least). The blue underline goes all over the place:
![screen shot 2017-01-02 at 5 35 20 pm](https://cloud.githubusercontent.com/assets/945577/21598474/92947b50-d112-11e6-81fc-38752109f23e.png)
The middle part that you fixed in ab337e2 seems fine.

11. Readme: Please remove the "Overview", "Why this new architecture?", "Warning", "Usage - templates and content", and "Usage - styling" sections. Add in a new Overview section which just says "This is a custom Open edX LMS theme (comprehensive theme) used by Cloudera. It is compatible with the Eucalyptus release of Open edX and newer." Mention how to build the CSS (presumably "make build" or do nothing? The content in "Usage -styling" seems out of date).

pomegranited commented:
@bradenmacdonald Thank you for your thorough review!

8. Ah oops.. I didn't notice that I'd increased the clickable area to the whole popup!  I've fixed it properly with 03c12c2.
9. Fixed with 8849a96.  Thanks for spotting that!
10. Fixed with a32d68e.  I'd seen that, but couldn't figure it out..  Turns out it's related to the problematic `button:not(:disabled)` CSS that affected everything under `.course-content`!  So I limited it part to just `.seq_content`.
11. Fixed with 74f3b47.  I've added the `make build` instructions to the `Theme Authoring` section.  We shouldn't have to do anything when installing the theme on a normal instance.

bradenmacdonald commented:
:+1:

- [x] I tested this: on the sandbox, extensively.
- [x] I read through the code
- [ ] I checked for accessibility issues: This theme has major a11y issues, but I'm going to consider those out of scope for this upgrade. I will open a Trello ticket.
- [x] Includes documentation: Minimal, but yes
